lib/pbench/agent/results.py

```diff
@@ -14,45 +14,79 @@
 from pbench.common.utils import md5sum, validate_hostname
 
 
-class FileUploadError(Exception):
-    """Raised when the uploading of file to server has failed"""
-
-    pass
-
-
 class MakeResultTb:
     """MakeResultTb - Creates the result tar file.
 
-        TODO:  This is latent code -- it is currently unused and largely
-               untested, intended to support the future implementation of
-               a tool to replace pbench-make-result-tb.
     """
 
+    class AlreadyCopied(RuntimeError):
+        """Specific run time error raised when a result directory has a .copied
+        marker present.
+        """
+
+        pass
+
+    class BenchmarkRunning(RuntimeError):
+        """Specific run time error raised when a result directory has a .running
+        marker present.
+        """
+
+        pass
+
     def __init__(
         self,
         result_dir: str,
         target_dir: str,
+        controller: str,
         config: PbenchAgentConfig,
         logger: Logger,
     ):
-        """__init__ - Initializes the required attributes
+        """__init__ - Initializes the required attributes for making a tar ball
 
-            Args:
-                result_dir -- directory where tb file is collected.
-                target_dir -- directory where tar file needs to be moved.
-                config -- PbenchAgent config object
-                logger -- logger objects helps logging important details
+        Args:
+            result_dir -- directory where tb file is collected.
+            target_dir -- directory where tar file needs to be moved.
+            config -- PbenchAgent config object
+            logger -- logger objects helps logging important details
+
+        Raises:
+            AlreadyCopied       if we find a "*.copied" marker for the result
+                                directory.
+            BenchmarkRunning    if we find a "*/.running" marker in the result
+                                directory.
+            FileNotFoundError   if either the result or target directories do
+                                not exist
+            NotADirectoryError  if either the result or target directories are
+                                not actual directories
+            RuntimeError        if it cannot find the 'tar' command on the PATH
```

Comment:
Not sure I follow.

Generic exception like `FileNotFoundError` and `NotADirectoryError` and `ValueError` are not self explanatory, and only when the exception happens will a string be provided to know what to do with it.

While working on this code, I found myself pouring over the existing to code find the behaviors on how to handle them in the "results move" loop, that it was easier to collect them once, then to keep reading the code as they were updated.

What is the harm in documenting them since they encode the behavior we are trying to implement?